rclpy/src/rclpy/client.cpp

```diff
@@ -0,0 +1,227 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+
+#include <memory>
+#include <string>
+
+extern "C"
+{
+#include "rclpy_common/common.h"
+}
+#include "rclpy_common/handle.h"
+
+#include "rclpy_common/exceptions.hpp"
+
+#include "client.hpp"
+
+namespace rclpy
+{
+static void
+_rclpy_destroy_client(void * p)
+{
+  auto cli = static_cast<rclpy_client_t *>(p);
+  if (!cli) {
+    // Warning should use line number of the current stack frame
```

Comment:
There was a `PyErr_Clear()` in the `_rclpy.c` code. I assume it was removed on purpose?

Reply:
It doesn't really make sense to me why that was there in the first place. None of the other `_rclpy_destroy_*` functions have it.